### PR TITLE
Zuri training

### PR DESCRIPTION
**Pull request by robertechie**
# ENDPOINT: localhost:5600/api/v1/zuri/training/application
# application type: application/json
form method: POST
form fields:
     "firstName":"ROBERT",
      "lastName":"ERU",
      "email":"sumeyuu@gmail.com",
      "country":"Nigeria",
      "state":"Delta",
      "track":"frontend", // frontend', 'mobile', 'design', 'backend
      "employmentStatus":"employed",  // 'employed', 'unemployed', 'student'
      "gender":"male",   // male', 'female
      "dob":"2020/01/01",
      "phoneNumber":"4569858698"

Capture

### CODE REVIEW

File: config.js [hunk 1]
```diff
@@ -4,6 +4,6 @@ module.exports = {
   dbPass: process.env.PASS || 'hngi8be',
   dbname: process.env.NAME || 'hngi8be',
   // dbconnection: process.env.DBURL || 'mongodb+srv://hngi8be:hngi8be@hngi8.qqarl.mongodb.net/hngi8be?retryWrites=true&w=majority',
-  dbconnection: process.env.DBURL || 'mongodb+srv://tmayor20:tmayor2020@cluster0.tui54.mongodb.net/tmayor20?retryWrites=true&w=majority',
+  dbconnection: process.env.DBURL || 'mongodb+srv://Roberto:freeme2018@cluster0-xdavb.mongodb.net/test?retryWrites=true&w=majority',
```
Comment (tonymj76) [inline config.js]: why change this?

File: controller/zuriTrainingApplicationController.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
+/* eslint-disable new-cap */
+/* eslint-disable no-console */
+const { body, validationResult } = require('express-validator');
+const Intern = require('../models/zuriTrainingModel');
```
Comment (tonymj76) [inline controller/zuriTrainingApplicationController.js]: change the model to ZuriTrainingModel

File: controller/zuriTrainingApplicationController.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  body('track').isString().not().isEmpty(),
+  body('employmentStatus').isString().not().isEmpty(),
+  body('gender').isString().not().isEmpty(),
+  body('dob').isString().not().isEmpty(),
+  body('phoneNumber').isLength({ min: 10 }).not().isEmpty()
+
+];
+const createIntern = async (req, res) => {
+  const {
+    firstName, lastName, email, country, state, track, employmentStatus, gender, dob, phoneNumber
```
Comment (tonymj76) [inline controller/zuriTrainingApplicationController.js:22]: what is dob

File: controller/zuriTrainingApplicationController.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        message: 'Error occured creating an intern profile'
+      });
+    }
+
+    return res.status(200).json({
+      status: 'success',
+      message: 'Application Successfully Registered'
+    });
+  } catch (error) {
+    return res.status(500).json({
```
Comment (tonymj76) [inline controller/zuriTrainingApplicationController.js]: change the response body to the general format

File: config.js (hunk 1 above)
Comment (tonymj76) [inline config.js]: why change the db url

File: controller/zuriTrainingApplicationController.js (hunk 2 above)
Comment (tonymj76) [inline controller/zuriTrainingApplicationController.js]: change the model to the proper one

File: controller/zuriTrainingApplicationController.js [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+  try {
+    const errors = validationResult(req);
+    const err = errors.array();
+    const myarray = [];
+    err.forEach((er) => {
+      const message = `${er.msg} in ${er.param}`;
+      myarray.push(message);
+    });
+    if (myarray.length > 0) {
+      return res.status(422).json({
```
Comment (tonymj76) [inline controller/zuriTrainingApplicationController.js]: change the the general response handler

File: controller/zuriTrainingApplicationController.js [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+      });
+    }
+  } catch (error) {
+    return false;
+  }
+
+  try {
+    const checkIntern = await Intern.findOne({ email });
+    if (checkIntern) {
+      return res.status(200).json({
```
Comment (tonymj76) [inline controller/zuriTrainingApplicationController.js]: change

**Review by tonymj76 — CHANGES_REQUESTED** [review]

File: controller/zuriTrainingApplicationController.js (hunk 2 above)
Comment (robertechie) [reply controller/zuriTrainingApplicationController.js]: that what i found there

**Review by tonymj76 — APPROVED** [review]

**Pull request merged**